devicetypes/samsungsds/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy

```diff
@@ -0,0 +1,412 @@
+	sendEvent(name: "lock", value: "unlocked", isStateChange: true, displayed: false)
+	sendEvent(name: "battery", value: "100", isStateChange: true, displayed: false)
+	// Device-Watch allows 2 check-in misses from device + ping (plus 2 min lag time)
+	// Currently set for testing ping interval set to 3 min
+	sendEvent(name: "checkInterval", value: 60 + 2 * 60, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+
+	log.info "ZigBee DTH - configure() returning with cmds:- $cmds"
+	cmds
+}
+
+
+/**
+ * Executes unlock command on a Zigbee lock
+ */
+def unlock() {
+	log.trace "ZigBee DTH - Executing unlock() for device ${device.displayName}"
+	def cmds = zigbee.command(CLUSTER_DOORLOCK, DOORLOCK_CMD_UNLOCK_DOOR, "100431323335", [mfgCode: 0003])
+	log.info "ZigBee DTH - unlock() returning with cmds:- $cmds"
+	return cmds
+}
+
+
+/**
+ * Responsible for parsing incoming device messages to generate events
+ *
+ * @param description The incoming description from the device
+ *
+ * @return result: The list of events to be sent out
+ *
+ */
+def parse(String description) {
+	log.trace "ZigBee DTH - Executing parse() for device ${device.displayName}"
+	def result = null
+	if (description) {
+		if (description.startsWith('read attr -')) {
+			result = parseAttributeResponse(description)
+		} else {
+			result = parseCommandResponse(description)
+		}
+	}
+	return result
+}
+
+/**
+ * Responsible for handling attribute responses
+ *
+ * @param description The description to be parsed
+ *
+ * @return result: The list of events to be sent out
+ */
+private def parseAttributeResponse(String description) {
+	Map descMap = zigbee.parseDescriptionAsMap(description)
+	log.trace "ZigBee DTH - Executing parseAttributeResponse() for device ${device.displayName} with description map:- $descMap"
+	def result = []
+	Map responseMap = [:]
+	def clusterInt = descMap.clusterInt
+	def attrInt = descMap.attrInt
+	def deviceName = device.displayName
+	if (clusterInt == CLUSTER_POWER && attrInt == POWER_ATTR_BATTERY_VOLTAGE) {
+		responseMap.name = "battery"
+		responseMap.value = getBatteryResult(Integer.parseInt(descMap.value, 10))
+		responseMap.descriptionText = "Battery is at ${responseMap.value}%"
+	} else if (clusterInt == CLUSTER_DOORLOCK && attrInt == DOORLOCK_ATTR_LOCKSTATE) {
+		def value = Integer.parseInt(descMap.value, 16)
+		responseMap.name = "lock"
+		if (value == 0) {
+			responseMap.value = "unknown"
+			responseMap.descriptionText = "Unknown state"
+		} else if (value == 1) {
+			responseMap.value = "locked"
+			responseMap.descriptionText = "Locked"
+		} else if (value == 2) {
+			responseMap.value = "unlocked"
+			responseMap.descriptionText = "Unlocked"
+		} else {
+			responseMap.value = "unknown"
+			responseMap.descriptionText = "Unknown state"
+		}
+	} else {
+		log.trace "ZigBee DTH - parseAttributeResponse() - ignoring attribute response"
+		return null
+	}
+
+	if (responseMap.data) {
```

Comment:
No code in before here will set `responseMap.data` so I think you can remove the `if` and just leave the code from the `else`